app/models/concerns/course/assessment/questions_concern.rb

```diff
@@ -22,6 +22,14 @@ def attempt(submission)
     end
   end
 
+  # Returns the questions which do not have a answer.
+  #
+  # @param [Course::Assessment::Submission] submission The submission which contains the answers.
+  # @return [Array<Course::Assessment::Question>]
+  def not_answered(submission)
+    where.not(id: submission.answers.select(:question_id))
```

Comment:
This is different, that helper is for getting correctly answered question ids. but here this method returns the questions which do not have an answer.